superset-frontend/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -16,46 +16,122 @@
+      expect(screen.getByText('Checkbox Label')).toBeInTheDocument();
+    });
+
+    test('should render the checkbox', async () => {
+      await asyncRender();
+      expect(screen.getByRole('checkbox')).toBeInTheDocument();
+    });
   });
-});
 
-test('works with an onChange handler', () => {
-  const mockAction = jest.fn();
-  const { getByRole } = render(
-    <Checkbox style={{}} checked={false} onChange={mockAction} />,
-  );
-  fireEvent.click(getByRole('checkbox'));
-  expect(mockAction).toHaveBeenCalled();
-});
+  describe('States', () => {
+    test('should render as unchecked when checked is false', async () => {
+      await asyncRender();
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).not.toBeChecked();
+    });
+
+    test('should render as checked when checked is true', async () => {
+      const checkedProps = { ...mockedProps, checked: true };
+      await asyncRender(checkedProps);
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).toBeChecked();
+    });
+
+    test('should render as indeterminate when indeterminate is true', async () => {
+      const indeterminateProps = { ...mockedProps, indeterminate: true };
+      await asyncRender(indeterminateProps);
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).toBeInTheDocument();
+      expect((checkbox as HTMLInputElement).indeterminate).toBe(true);
+    });
+
+    test('should render as disabled when disabled prop is true', async () => {
+      const disabledProps = { ...mockedProps, disabled: true };
+      await asyncRender(disabledProps);
+      const checkboxWrapper = screen
+        .getByRole('checkbox')
+        .closest('.ant-checkbox-wrapper');
+      expect(checkboxWrapper).toHaveClass('ant-checkbox-wrapper-disabled');
+    });
+  });
 
-test('renders custom Checkbox styles without melting', () => {
-  const { getByRole } = render(
-    <Checkbox onChange={() => true} checked={false} style={{ opacity: 1 }} />,
-  );
-  expect(getByRole('checkbox')).toBeInTheDocument();
-  expect(getByRole('checkbox')).toHaveStyle({ opacity: 1 });
+  describe('Interactions', () => {
+    test('should call the onChange handler when clicked', async () => {
+      await asyncRender();
+      const checkbox = screen.getByRole('checkbox');
+      await userEvent.click(checkbox);
+      expect(mockedProps.onChange).toHaveBeenCalledTimes(1);
+    });
+
+    test('should not call the onChange handler when disabled and clicked', async () => {
+      const mockOnChange = jest.fn();
+      const disabledProps = {
+        ...mockedProps,
+        disabled: true,
+        onChange: mockOnChange,
+      };
+
+      const { getByRole } = render(<Checkbox {...disabledProps} />);
+      const checkbox = getByRole('checkbox');
+
+      fireEvent.click(checkbox);
+
+      expect(mockOnChange).not.toHaveBeenCalled();
+    });
+
+    test('calls onChange handler with fireEvent API', () => {
```

Comment:
Not sure what is the purpose of specifying the fireEvent API in this case? Let's name the tests in a way that makes sense for a user behavior. 

```suggestion
    test('calls onChange handler successfully', () => {
```